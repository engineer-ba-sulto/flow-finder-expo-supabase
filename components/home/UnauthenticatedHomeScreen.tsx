import React from "react";
import { ScrollView, Text, View } from "react-native";
import { Button } from "../../components/ui/Button";
import { APP_DESCRIPTION } from "../../constants/app";

interface UnauthenticatedHomeScreenProps {
  router: any;
}

const UnauthenticatedHomeScreen: React.FC<UnauthenticatedHomeScreenProps> = ({
  router,
}) => (
  <ScrollView
    className="flex-1 bg-white"
    showsVerticalScrollIndicator={false}
    accessibilityLabel="ログイン前ホーム画面"
  >
    <View className="flex-1 p-6 justify-between">
      <View className="items-center">
        {/* メインアイコン */}
        <Text className="text-6xl mb-4" accessibilityLabel="Flow Finderアイコン">
          🎯
        </Text>
        
        {/* アプリタイトル */}
        <Text
          className="text-2xl font-bold text-center mb-4 text-gray-800"
          accessibilityRole="header"
          accessibilityLabel="Flow Finder"
        >
          Flow Finder
        </Text>
        
        {/* アプリ説明 */}
        <Text
          className="text-center text-gray-800 text-sm leading-relaxed mb-6"
          accessibilityLabel={APP_DESCRIPTION}
        >
          {APP_DESCRIPTION}
        </Text>

        {/* 機能紹介セクション */}
        <View className="bg-gray-50 rounded-xl p-6 mb-6 w-full">
          <Text
            className="text-sm font-medium text-gray-800 mb-3 text-center"
            accessibilityLabel="Flow Finderができること"
          >
            Flow Finderができること
          </Text>
          <View className="gap-3">
            <View className="flex-row items-center">
              <Text className="text-lg mr-3">🎯</Text>
              <View className="flex-1">
                <Text className="text-xs font-medium">ゴール設定</Text>
                <Text className="text-xs text-gray-600">目標を明確にする</Text>
              </View>
            </View>
            <View className="flex-row items-center">
              <Text className="text-lg mr-3">📊</Text>
              <View className="flex-1">
                <Text className="text-xs font-medium">ゴール管理</Text>
                <Text className="text-xs text-gray-600">目標を整理・管理</Text>
              </View>
            </View>
            <View className="flex-row items-center">
              <Text className="text-lg mr-3">✅</Text>
              <View className="flex-1">
                <Text className="text-xs font-medium">簡単管理</Text>
                <Text className="text-xs text-gray-600">シンプルな操作で継続</Text>
              </View>
            </View>
          </View>
        </View>

        {/* MVP1段階での機能説明 */}
        <View className="bg-blue-50 border border-blue-200 rounded-xl p-4 mb-6 w-full">
          <Text
            className="text-xs text-blue-800 font-medium mb-2"
            accessibilityLabel="MVP1段目：基本機能"
          >
            🚀 MVP1段目：基本機能
          </Text>
          <Text className="text-xs text-blue-600">
            • ゴール設定・管理{`\n`}
            • シンプルな管理{`\n`}
            • 簡単な完了機能
          </Text>
        </View>
      </View>

      {/* ボタンセクション */}
      <View className="w-full gap-3">
        <Button
          variant="primary"
          onPress={() => router.push("/auth/signup")}
          accessibilityLabel="今すぐ始める"
          accessibilityHint="新しいアカウントを作成して始めます"
        >
          今すぐ始める
        </Button>
        <Button
          variant="secondary"
          onPress={() => router.push("/auth/login")}
          accessibilityLabel="ログイン"
          accessibilityHint="既存のアカウントでログインします"
        >
          ログイン
        </Button>
      </View>
    </View>
  </ScrollView>
);

export default UnauthenticatedHomeScreen;
