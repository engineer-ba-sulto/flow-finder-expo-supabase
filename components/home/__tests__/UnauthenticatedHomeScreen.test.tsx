import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import UnauthenticatedHomeScreen from '../UnauthenticatedHomeScreen';

// モック設定
const mockRouter = {
  push: jest.fn(),
  replace: jest.fn(),
  back: jest.fn(),
};

jest.mock('../../../components/ui/Button', () => {
  const { Text, TouchableOpacity } = require('react-native');
  return {
    Button: ({ children, onPress, testID, className }: any) => (
      <TouchableOpacity onPress={onPress} testID={testID} className={className}>
        <Text>{children}</Text>
      </TouchableOpacity>
    ),
  };
});

jest.mock('../../../constants/app', () => ({
  APP_DESCRIPTION: 'あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ',
}));

describe('UnauthenticatedHomeScreen', () => {
  const defaultProps = {
    router: mockRouter,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('基本表示', () => {
    it('メインウェルカムメッセージが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('Flow Finderへようこそ')).toBeTruthy();
    });

    it('アプリ説明が表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ')).toBeTruthy();
    });

    it('特徴紹介メッセージが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText(/成長の障壁を特定し、具体的な解決策を提案する/)).toBeTruthy();
      expect(screen.getByText(/パーソナルコーチングアプリです/)).toBeTruthy();
    });

    it('ログイン案内セクションが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('ログインして始めましょう')).toBeTruthy();
    });

    it('ログインボタンが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('ログイン')).toBeTruthy();
    });

    it('新規登録ボタンが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('新規登録')).toBeTruthy();
    });

    it('無料案内メッセージが表示される', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByText('無料で始められます')).toBeTruthy();
    });
  });

  describe('ナビゲーション', () => {
    it('ログインボタンをタップするとログイン画面に遷移する', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      const loginButton = screen.getByText('ログイン');
      fireEvent.press(loginButton);
      
      expect(mockRouter.push).toHaveBeenCalledWith('/auth/login');
    });

    it('新規登録ボタンをタップすると登録画面に遷移する', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      const signupButton = screen.getByText('新規登録');
      fireEvent.press(signupButton);
      
      expect(mockRouter.push).toHaveBeenCalledWith('/auth/signup');
    });

    it('ボタンを複数回タップしても正しく動作する', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      const loginButton = screen.getByText('ログイン');
      const signupButton = screen.getByText('新規登録');
      
      fireEvent.press(loginButton);
      fireEvent.press(signupButton);
      fireEvent.press(loginButton);
      
      expect(mockRouter.push).toHaveBeenCalledTimes(3);
      expect(mockRouter.push).toHaveBeenNthCalledWith(1, '/auth/login');
      expect(mockRouter.push).toHaveBeenNthCalledWith(2, '/auth/signup');
      expect(mockRouter.push).toHaveBeenNthCalledWith(3, '/auth/login');
    });
  });

  describe('レイアウトとスタイル', () => {
    it('ScrollViewが正しく設定されている', () => {
      const { toJSON } = render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      // ScrollViewの構造を確認
      expect(toJSON()).toMatchSnapshot();
    });

    it('適切な背景色とパディングが設定されている', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      // 要素が存在することでスタイルが正しく適用されていることを確認
      expect(screen.getByText('Flow Finderへようこそ')).toBeTruthy();
    });
  });

  describe('アクセシビリティ', () => {
    it('適切なアクセシビリティラベルが設定されている', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByLabelText('ログイン案内画面')).toBeTruthy();
      expect(screen.getByLabelText('Flow Finderへようこそ')).toBeTruthy();
      expect(screen.getByLabelText('ログイン画面に移動')).toBeTruthy();
      expect(screen.getByLabelText('新規登録画面に移動')).toBeTruthy();
      expect(screen.getByLabelText('無料で始められます')).toBeTruthy();
    });

    it('ヘッダーロールが正しく設定されている', () => {
      const { toJSON } = render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      // accessibilityRoleがheaderに設定されていることを確認
      expect(toJSON()).toMatchSnapshot();
    });

    it('アクセシビリティヒントが設定されている', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      expect(screen.getByLabelText('既存のアカウントでログインします')).toBeTruthy();
      expect(screen.getByLabelText('新しいアカウントを作成します')).toBeTruthy();
    });
  });

  describe('コンテンツのバリエーション', () => {
    it('APP_DESCRIPTIONが異なる値でも正しく表示される', () => {
      // 一時的にモックを変更
      jest.doMock('../../../constants/app', () => ({
        APP_DESCRIPTION: '別のアプリ説明文',
      }));
      
      // コンポーネントを再度要求
      const { UnauthenticatedHomeScreen: NewComponent } = require('../UnauthenticatedHomeScreen');
      
      render(<NewComponent {...defaultProps} />);
      
      // 新しい説明文が表示されることを確認
      expect(screen.getByText('別のアプリ説明文')).toBeTruthy();
    });
  });

  describe('コンポーネントの構造', () => {
    it('正しい階層構造でレンダリングされる', () => {
      const { toJSON } = render(<UnauthenticatedHomeScreen {...defaultProps} />);
      const tree = toJSON();
      
      // ルート要素がScrollViewであることを確認
      expect(tree?.type).toBe('ScrollView');
    });

    it('必要なセクションがすべて含まれている', () => {
      render(<UnauthenticatedHomeScreen {...defaultProps} />);
      
      // メインウェルカムセクション
      expect(screen.getByText('Flow Finderへようこそ')).toBeTruthy();
      
      // 特徴紹介
      expect(screen.getByText(/成長の障壁を特定し/)).toBeTruthy();
      
      // ログイン案内セクション
      expect(screen.getByText('ログインして始めましょう')).toBeTruthy();
      
      // ボタン
      expect(screen.getByText('ログイン')).toBeTruthy();
      expect(screen.getByText('新規登録')).toBeTruthy();
      
      // 追加情報
      expect(screen.getByText('無料で始められます')).toBeTruthy();
    });
  });

  describe('プロパティの処理', () => {
    it('routerプロパティが正しく渡される', () => {
      const customRouter = {
        push: jest.fn(),
        replace: jest.fn(),
        back: jest.fn(),
      };
      
      render(<UnauthenticatedHomeScreen router={customRouter} />);
      
      const loginButton = screen.getByText('ログイン');
      fireEvent.press(loginButton);
      
      expect(customRouter.push).toHaveBeenCalledWith('/auth/login');
    });
  });
});