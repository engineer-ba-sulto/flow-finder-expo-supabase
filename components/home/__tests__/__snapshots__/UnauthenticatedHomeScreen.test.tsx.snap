// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthenticatedHomeScreen アクセシビリティ ヘッダーロールが正しく設定されている 1`] = `
<RCTScrollView
  accessibilityLabel="ログイン案内画面"
  className="flex-1 bg-gray-50"
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      className="flex-1 px-4 pt-12 pb-6"
    >
      <View
        className="bg-white rounded-xl p-6 mb-6 shadow-sm"
      >
        <Text
          accessibilityLabel="Flow Finderへようこそ"
          accessibilityRole="header"
          className="text-3xl font-bold text-center mb-4 text-gray-800"
        >
          Flow Finderへようこそ
        </Text>
        <Text
          accessibilityLabel="あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ"
          className="text-center text-gray-600 mb-6 leading-6 text-base"
        >
          あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ
        </Text>
        <View
          className="bg-blue-50 rounded-lg p-4"
        >
          <Text
            accessibilityLabel="成長の障壁を特定し、具体的な解決策を提案するパーソナルコーチングアプリです"
            className="text-center text-blue-700 font-medium"
          >
            成長の障壁を特定し、具体的な解決策を提案する
            

            パーソナルコーチングアプリです
          </Text>
        </View>
      </View>
      <View
        className="bg-white rounded-xl p-6 shadow-sm"
      >
        <Text
          accessibilityRole="header"
          className="text-xl font-semibold text-center mb-6 text-gray-800"
        >
          ログインして始めましょう
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text>
            ログイン
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text>
            新規登録
          </Text>
        </View>
        <Text
          accessibilityLabel="無料で始められます"
          className="text-center text-gray-500 text-sm mt-4"
        >
          無料で始められます
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`UnauthenticatedHomeScreen レイアウトとスタイル ScrollViewが正しく設定されている 1`] = `
<RCTScrollView
  accessibilityLabel="ログイン案内画面"
  className="flex-1 bg-gray-50"
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      className="flex-1 px-4 pt-12 pb-6"
    >
      <View
        className="bg-white rounded-xl p-6 mb-6 shadow-sm"
      >
        <Text
          accessibilityLabel="Flow Finderへようこそ"
          accessibilityRole="header"
          className="text-3xl font-bold text-center mb-4 text-gray-800"
        >
          Flow Finderへようこそ
        </Text>
        <Text
          accessibilityLabel="あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ"
          className="text-center text-gray-600 mb-6 leading-6 text-base"
        >
          あなたの成長を妨げる「見えない壁」を見つけ、壊すためのパーソナルコーチング アプリ
        </Text>
        <View
          className="bg-blue-50 rounded-lg p-4"
        >
          <Text
            accessibilityLabel="成長の障壁を特定し、具体的な解決策を提案するパーソナルコーチングアプリです"
            className="text-center text-blue-700 font-medium"
          >
            成長の障壁を特定し、具体的な解決策を提案する
            

            パーソナルコーチングアプリです
          </Text>
        </View>
      </View>
      <View
        className="bg-white rounded-xl p-6 shadow-sm"
      >
        <Text
          accessibilityRole="header"
          className="text-xl font-semibold text-center mb-6 text-gray-800"
        >
          ログインして始めましょう
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text>
            ログイン
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text>
            新規登録
          </Text>
        </View>
        <Text
          accessibilityLabel="無料で始められます"
          className="text-center text-gray-500 text-sm mt-4"
        >
          無料で始められます
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
